ui/src/app/pages/admin/workspace/cloud-environments-table.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import { useState } from 'react';
+import { Column } from 'primereact/column';
+import { DataTable } from 'primereact/datatable';
+
+import {
+  AdminWorkspaceResources,
+  ListRuntimeResponse,
+  UserAppEnvironment,
+} from 'generated/fetch';
+
+import {
+  fromRuntimeStatus,
+  fromUserAppStatusWithFallback,
+  toUIAppType,
+  UIAppType,
+} from 'app/components/apps-panel/utils';
+import { Button } from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import { TooltipTrigger } from 'app/components/popups';
+import { workspaceAdminApi } from 'app/services/swagger-fetch-clients';
+import { getCreator } from 'app/utils/runtime-utils';
+
+interface CloudEnvironmentRow {
```

Comment:
"row" here means a row of this table.  Is that obvious, or would a comment or rename help here?

Reply:
I think it's obvious since it's in the context of a file titled `cloud-environments-table.tsx`.